src/bindings/python/src/pyopenvino/core/tensor.cpp

```diff
@@ -13,12 +13,23 @@
 
 namespace py = pybind11;
 
+class TensorWrapper {
+public:
+    TensorWrapper(ov::Tensor&& tensor, py::object array) : m_tensor{std::move(tensor)}, array_handle{array} {}
+
+    TensorWrapper(ov::Tensor&& tensor) : m_tensor{std::move(tensor)} {}
+
+    ov::Tensor m_tensor;
+    py::object array_handle;  // Holds the Python array to manage its lifetime
+};
+
 void regclass_Tensor(py::module m) {
-    py::class_<ov::Tensor, std::shared_ptr<ov::Tensor>> cls(m, "Tensor");
+    py::class_<TensorWrapper, std::shared_ptr<TensorWrapper>> cls(m, "Tensor");
     cls.doc() = "openvino.runtime.Tensor holding either copy of memory or shared host memory.";
 
     cls.def(py::init([](py::array& array, bool shared_memory) {
-                return Common::object_from_data<ov::Tensor>(array, shared_memory);
+                auto tensor = Common::object_from_data<ov::Tensor>(array, shared_memory);
+                return TensorWrapper(std::move(tensor), std::move(array));
```

Comment:
1. `TensorWrapper(ov::Tensor&& tensor, py::object array)` doesn't take rvalue to `array`.
2. Given that this lambda takes `py::array& array` by ref, does moving from it affect the actual python object?

Reply:
it was draft with already outdated code.